drivers/sensor/nxp/nxp_tmpsns/nxp_tmpsns.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2025 Polytech A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_tmpsns
+
+#include <fsl_tempsensor.h>
+#include <zephyr/drivers/sensor.h>
+
+struct nxp_tmpsns_data {
+	float die_temp;
+};
+
+static int nxp_tmpsns_channel_get(const struct device *dev, enum sensor_channel chan,
+				  struct sensor_value *val)
+{
+	const struct nxp_tmpsns_data *data = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_DIE_TEMP:
+		return sensor_value_from_float(val, data->die_temp);
+	default:
+		return -ENOTSUP;
+	}
+}
+
+static int nxp_tmpsns_channel_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct nxp_tmpsns_data *data = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_ALL:
+	case SENSOR_CHAN_DIE_TEMP:
+		data->die_temp = TMPSNS_GetCurrentTemperature(TMPSNS);
+		return 0;
+	default:
+		return -ENOTSUP;
+	}
+}
+
+static DEVICE_API(sensor, tmpsns_driver_api) = {
+	.channel_get = nxp_tmpsns_channel_get,
+	.sample_fetch = nxp_tmpsns_channel_fetch,
+};
+
+static int nxp_tmpsns_init(const struct device *dev)
+{
+	tmpsns_config_t config;
+
+	TMPSNS_GetDefaultConfig(&config);
+	config.measureMode = kTEMPSENSOR_ContinuousMode;
+	config.frequency = 0x03U;
+	TMPSNS_Init(TMPSNS, &config);
+	TMPSNS_StartMeasure(TMPSNS);
```

Comment:
Is it necessary to start measuring on init? Why not enable the attribute for sample frequency to start sampling?

Reply:
Was done in init to allow parsing the `poll` part of the test in `tests/drivers/sensor/temp_sensor`